Dockerfile

```diff
@@ -3,12 +3,13 @@ FROM ubuntu:18.04
 ARG PREFIX=/usr/local
 ARG GPU_ARCH=";"
 ARG MIOTENSILE_VER="default"
+ARG USE_TARGETID="OFF"
 
 # Support multiarch
 RUN dpkg --add-architecture i386
 
 # Add rocm repository
-RUN sh -c 'echo deb [arch=amd64 trusted=yes] http://repo.radeon.com/rocm/apt/.apt_3.7/ xenial main > /etc/apt/sources.list.d/rocm.list'
+RUN if [ "$USE_TARGETID" = "ON" ] ; then sh -c 'echo deb [arch=amd64 trusted=yes] http://compute-artifactory.amd.com/artifactory/list/rocm-osdb-deb/ compute-rocm-dkms-no-npi-hipclang 6416 > /etc/apt/sources.list.d/rocm.list'; else sh -c 'echo deb [arch=amd64 trusted=yes] http://repo.radeon.com/rocm/apt/.apt_3.7/ xenial main > /etc/apt/sources.list.d/rocm.list'; fi
```

Comment:
[Q] Why 6416? Shall we promote this number in the future (and when)?

Reply:
Remember how long this PR has been staying here? :)
Let me know which mainline# is used currently

Reply:
Hmm... Do we want this to advance automatically, to the latest mainline build?

Reply:
What is the latest mainline build number?

Reply:
You can find it [here](http://confluence.amd.com/display/MLSE/Docker+images+for+Mainline+builds).

Reply:
But I do not know how good or bad is it ;)